lib/purchases_flutter.dart

```diff
@@ -633,6 +686,27 @@ class Purchases {
   /// from the billing services and clean up resources
   static Future<void> close() => _channel.invokeMethod('close');
 
+  /// iOS only. Starts the purchase flow associated with the callback at
+  /// `callbackID` index in PurchasesFlutterPlugin.m's `startPurchaseBlocks`
+  /// array.
+  static Future<PromotedPurchaseResult> _startPromotedProductPurchase(
+      int callbackID,) async {
+    final result = await _channel.invokeMethod(
+      'startPromotedProductPurchase',
+      {
+        'callbackID': callbackID,
+      },
+    );
+    final purchaserInfo = PurchaserInfo.fromJson(
+      Map<String, dynamic>.from(result['purchaserInfo']),
+    );
+    final productIdentifier = result['productIdentifier'];
+    return PromotedPurchaseResult(
+      productIdentifier: productIdentifier,
+      purchaserInfo: purchaserInfo,
+    );
```

Comment:
`_startPromotedProductPurchase` is private but did you try calling it from the tests? In dart, privacy is at library level (not class like in Java) so I think you might still be able to invoke it. I am not sure. 

It would actually be better if you find a way to invoke `'Purchases-ReadyForPromotedProductPurchase'` in the tests to fire the event and test that it does what it should. We don't have any example on how to do that but you have access to the channel in purchases_flutter_test.dart, so I am pretty sure it's probably something like `channel.invokeMethod('Purchases-ReadyForPromotedProductPurchase', arguments)`, same way you are doing in https://github.com/RevenueCat/purchases-flutter/pull/344/files#diff-a84fbe19c95708058ab36b565eb31eb4a3071f224431bdb07085001b0f6271bbR509. I haven't tested it though, so it's just an idea.

Reply:
yeah, it wouldn't let me call it from tests. i COULD make it public and then use @visibleForTesting as described [here](https://stackoverflow.com/a/54219882), but that would only warn if someone tried to call it from outside our library.

and invoking that `'Purchases-ReadyForPromotedProductPurchase'` event also wasn't working, i thought it should too since it's the same channel. i'll do a little more research on whether we can set that up...

Reply:
Alright I see your point. I don't want to block this PR on that, since we are missing a bunch of other unit tests. I tried and I also couldn't get it to work 🤔

Reply:
thanks for trying... i can make a sharpen the saw ticket for investigating further flutter unit tests